libscript/src/script-instance.cpp

```diff
@@ -673,61 +658,40 @@ static inline void MCScriptResolveDefinitionInFrame(MCScriptFrame *p_frame, uind
     r_definition = t_definition;
 }
 
-static inline MCValueRef MCScriptFetchFromLocalInFrame(MCScriptFrame *p_frame, uindex_t p_local)
+static inline MCTypeInfoRef MCScriptGetRegisterTypeInFrame(MCScriptFrame *p_frame, int p_register)
 {
-    /* LOAD CHECK */ __MCScriptAssert__(MCScriptIsLocalValidInFrame(p_frame, p_local),
-                                        "local out of range on fetch");
+    /* LOAD CHECK */ __MCScriptAssert__(MCScriptIsRegisterValidInFrame(p_frame, p_register),
+                                        "register out of range on fetch register type");
     
-    uindex_t t_slot_index;
-    t_slot_index = MCScriptComputeSlotIndexOfLocalInFrame(p_frame, p_local);
+    MCTypeInfoRef t_handler_type;
+    t_handler_type = p_frame -> instance -> module -> types[p_frame -> handler -> type] -> typeinfo;
     
-    return p_frame -> slots[t_slot_index];
-}
-
-static inline void MCScriptStoreToLocalInFrame(MCScriptFrame *p_frame, uindex_t p_local, MCValueRef p_value)
-{
-    /* LOAD CHECK */ __MCScriptAssert__(MCScriptIsLocalValidInFrame(p_frame, p_local),
-                                        "local out of range on store");
+    uindex_t t_parameter_count;
+    t_parameter_count = MCHandlerTypeInfoGetParameterCount(t_handler_type);
     
-    uindex_t t_slot_index;
-    t_slot_index = MCScriptComputeSlotIndexOfLocalInFrame(p_frame, p_local);
+    if (p_register < t_parameter_count)
+        return MCHandlerTypeInfoGetParameterType(t_handler_type, p_register);
     
-    if (p_frame -> slots[t_slot_index] != p_value)
-    {
-        MCValueRelease(p_frame -> slots[t_slot_index]);
-        p_frame -> slots[t_slot_index] = MCValueRetain(p_value);
-    }
+    if (p_register < t_parameter_count + p_frame -> handler -> local_type_count)
```

Comment:
The problem here is that the number of typed registers is a function of the parameter count and actual local variable count. The type information for these two things are stored in different places at the moment - and the parameter count is used to decide where. Making the latter check use a function or macro would require multiple evals of the parameter count which I'd like to avoid.